api/test/test_parse_parameter_name.py

```diff
@@ -0,0 +1,20 @@
+import pytest
+from dependencies import parse_parameter_name
+from fastapi import HTTPException
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "parameter_name_input, expected",
+    [
+        ("atemp:2.0:min:PT10M", HTTPException),
+        ("air_temperature:2.0m:min:PT10M", HTTPException),
+        ("air_temperature:2.0:product:PT10M", HTTPException),
+        ("air_temperature:2.0:min:P10", HTTPException),
+        (":", HTTPException),  # Too few parameters
+        ("::::::", HTTPException),
+    ],
+)  # Too many parameter
```

Comment:
The colons should be changed to underscores right?
```suggestion
@pytest.mark.parametrize(
    "parameter_name_input, expected",
    [
        ("atemp_2.0_min_PT10M", HTTPException),
        ("air_temperature_2.0m_min_PT10M", HTTPException),
        ("air_temperature_2.0_product_PT10M", HTTPException),
        ("air_temperature_2.0_min_P10", HTTPException),
        ("_", HTTPException),  # Too few parameters
        ("______", HTTPException),
    ],
)  # Too many parameter
```

Reply:
No, this feature was dropped. Have removed the test.